### PR TITLE
Fix y-axis range of main plot in dataviewer from 0 to 1 when P1 data type

### LINKED ISSUE #225: Fix y-axis range of main plot in dataviewer from 0 to 1 when P1 data type
### Feature you want to implement
According to requests by an experimenter, I will fix the y-axis range of main plot in dataviewer app from 0 to 1, when the current data type is P1.

### How the feature is implemented
Pass the data type to `MainPlotWidget.setData()`.

### PR DESCRIPTION
**Pull request by BECATRUE**
This closes #225.

Now, once the P1 data type button is clicked, the y-axis range is set from 0 to 1.

image

### CODE REVIEW

File: iquip/apps/dataviewer.py [hunk 1]
```diff
@@ -691,6 +691,7 @@ def _idToggledSlot(self, id_: int, checked: bool):
             checked: Whether the button is now checked or not.
         """
         if checked:
+            self.buttonGroup.button(id_).setChecked(True)
```
Comment (kangz12345) [inline iquip/apps/dataviewer.py]: What is this for?

File: iquip/apps/dataviewer.py [hunk 2]
```diff
@@ -1157,7 +1163,7 @@ def updateMainPlot(self, axis: Sequence[int], dataType: DataPointWidget.DataType
             return
         reduce = self._reduceFunction(dataType)
         data, axes = self.policy.extract(axis, reduce)
-        self.frame.mainPlotWidget.setData(data, axes)
+        self.frame.mainPlotWidget.setData(data, axes, dataType)
```
Comment (kangz12345) [inline iquip/apps/dataviewer.py]: I think we should `setYRange()` here, i.e., `self.frame.mainPlotWidget.viewer().plotItem.setYRange(0, 1)`,
since the `MainPlotWidget` is independent to the `DataType`.
What do you think of this?

**Review by kangz12345 — CHANGES_REQUESTED** [review]

File: iquip/apps/dataviewer.py (hunk 1 above)
Comment (BECATRUE) [reply iquip/apps/dataviewer.py]: I thought I needed this when I was testing before, but when I looked back, I don't think so.
Please ignore it😅

File: iquip/apps/dataviewer.py (hunk 2 above)
Comment (BECATRUE) [reply iquip/apps/dataviewer.py]: I see! I updated it as you suggested.

File: iquip/apps/dataviewer.py [hunk 3]
```diff
@@ -762,11 +762,11 @@ def setData(self, data: np.ndarray, axes: Sequence[AxisInfo]):
             logger.error("MainPlotWidget does not support %d-dim data", data.ndim)
             return
         self.viewers[plotType].setData(data, axes)
-        self.stack.setCurrentIndex(plotType)
         if self.autoRangeBox.isChecked():
             plotItem = self.viewers[plotType].plotItem
             bounds = plotItem.getViewBox().childrenBoundingRect(items=None)
             plotItem.setXRange(bounds.left(), bounds.right())
+        self.stack.setCurrentIndex(plotType)
```
Comment (kangz12345) [inline iquip/apps/dataviewer.py:769]: I think this line is just moved from line 765.
If it is not intended, please move it back to line 765, since it makes an unnecessary diff.

**Review by kangz12345 — CHANGES_REQUESTED** [review]

File: iquip/apps/dataviewer.py (hunk 3 above)
Comment (BECATRUE) [reply iquip/apps/dataviewer.py:769]: As I think, it looks better to show the plot after all modifications are done.
What do you think of it?

Comment (kangz12345) [reply iquip/apps/dataviewer.py:769]: Oh I agree with that! Let's go with your decision.

**Review by kangz12345 — APPROVED** [review]
